checker/src/org/checkerframework/checker/guieffect/GuiEffectVisitor.java

```diff
@@ -156,7 +156,9 @@ public Void visitMethodInvocation(MethodInvocationTree node, Void p) {
         }
 
         Effect callerEffect = atypeFactory.getDeclaredEffect(callerElt);
-        assert (callerEffect.equals(effStack.peek()));
+        // Field initializers inside anonymous inner classes show up with a null current-method ---
+        // the traversal goes straight from the class to the initializer.
+        assert (currentMethods.peek() == null || callerEffect.equals(effStack.peek()));
```

Comment:
We do; ```currentMethods``` is a stack of ```MethodTree```s, where ```null``` is used as a sentinel to indicate that there is no surrounding method context.  Previously I thought this only applied to static initializers (which is why the sentinel exists in the first place), but apparently instance initializers also show up in this context.

If there's a more idiomatic way to handle the sentinel (some kind of non-null dummy method tree) I'm open to a bit of refactoring.